jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojo.java

```diff
@@ -0,0 +1,62 @@
+package com.google.cloud.tools.jib.maven.skaffold;
+
+import com.google.cloud.tools.jib.filesystem.TempDirectoryProvider;
+import com.google.cloud.tools.jib.maven.JibPluginConfiguration;
+import com.google.cloud.tools.jib.maven.MavenProjectProperties;
+import com.google.cloud.tools.jib.maven.MavenRawConfiguration;
+import com.google.cloud.tools.jib.maven.MojoCommon;
+import com.google.cloud.tools.jib.plugins.common.ContainerizingMode;
+import com.google.cloud.tools.jib.plugins.common.InvalidContainerizingModeException;
+import com.google.cloud.tools.jib.plugins.common.PluginConfigurationProcessor;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+@Mojo(
+    name = SyncMapMojo.GOAL_NAME,
+    requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class SyncMapMojo extends JibPluginConfiguration {
+
+  @VisibleForTesting static final String GOAL_NAME = "_skaffold-sync-map";
+
+  @Override
+  public void execute() throws MojoExecutionException {
+    checkJibVersion();
+    if (MojoCommon.shouldSkipJibExecution(this)) {
+      return;
+    }
+
+    // add check that means this is only for jars
+    if (!"jar".equals(getProject().getPackaging())) {
+      throw new MojoExecutionException("Sync is currently only available for jar style projects");
```

Comment:
If an error ever happens with Skaffold (and someone will hit some problem), I'd like to make the exception/log message clearly convey who was at fault and where exactly.

`"smart(?) Skaffold sync is currently only available for jar style Jib projects, but the packaging of " + project.getArtifactId() (or path or whatever) + " is " + project.getPackaging()`

And should we fall back to just not doing advanced sync when these kinds of checks fail? For example, they could have configured manual sync for some extra directory files, and they would continue to work regardless of whether or not it is JAR project or the exploded mode after we log a warning? (But then, I guess this smart sync is incompatible with other sync modes and this MOJO is called only when Skaffold determined it would to the smart sync? But even in that case, we may just give up syncing after printing a warning instead of just blocking using Skaffold? I don't know really.)

Reply:
I think our base case is that we only allow one type of sync for now. But I'll update the message.